docs/DefiningLabels.md

```diff
@@ -0,0 +1,123 @@
+
+### Status (`s.`)
+
+* Open issues
+    * No status: New issue yet to be triaged. 
+    * `s.Ongoing` : The issue is being worked on.
+* Open PR  
+    * `s.ToReview`: Waiting for the review.
+    * `s.Ongoing` : The PR is being worked on.
+    * `s.ToMerge`: Main reviewer approved the changes. 
+    * `s.MergeApproved` : Both main reviewer and the code quality reviewer has approved the merge. PR can be merged.
+    * `s.OnHold` : The work on the PR has been put on hold pending some other event.
+* Closed issue/PR
+    * `s.Discarded` : The PR was abandoned. 
+
+### Category (`c.`)
+
+* Changes to _functionality_, categorized based on size
+  * `c.Enhancement`: An enhancement to an existing functionality (not big enough 
+   to be considered as a user story).
+  * `c.Story`: A user story.
+  * `c.Epic`: A feature that is worth many user stories.
+* Other work
+  * `c.Bug`
+  * `c.Task`: Other work items such as updating documentation.
+  * `c.Message`: Issue used as a means of discussing something with the dev team.
+     e.g. a request for help on setting up dev environment.
+
+### Difficulty (`d.`)
+
+* `d.FirstTimers`: Easy. To do as the first issue for new developers. One developer
+  should not do more than one of these. Not to be applied for issues with a priority `p.high` or above.
+* `d.Contributors`: Moderate difficulty. Small localized change. Can be done by contributors.
+  Not to be applied for issues with a priority `p.high` or above.
+* `d.Committers`: More difficult issues that are better left for committers or more senior developers.
+
+### Effort (`e.`)
+
+This label is used to indicate how much effort is expected for (or was spent on)
+an issue.
+
+`e.1` is roughly equal to an hour of work, `e.2` is two hours of work, and so on.
+Recommended values: 1,2,4,8,16,32
+
+Effort labels applied to PRs indicate effort for reviewing (by the main reviewer). Those applied for issues indicate
+the effort for fixing the issue.
+
+### Aspect (`a-`)
+
+Classifies the issues based on the non-functional aspect it tackles. 
+
+|Label              | Description
+|-------------------|------------------------------------
+|`a-AccessControl`  |Controlling access to user groups, authentication, privacy, anonymity
+|`a-CodeQuality`    |Refactorings that are mainly to improve code/design quality
+|`a-Concurrency`    |Things related to concurrent access, session control
+|`a-DevOps`         |CI, release management, version control, dev docs
+|`a-Docs`           |Website, user docs
+|`a-FaultTolerance` |Resilience to user errors, environmental problems
+|`a-Performance`    |Speed of operation
+|`a-Persistence`    |Database layer, GAE datastore
+|`a-Scalability`    |Related to behavior at increasing loads
+|`a-Security`       |Protection from security threats
+|`a-Testing`        |Testing efficiency and robustness (as opposed to testing a specific feature)
+|`a-UIX`            |User interface, User experience, Responsiveness
```

Comment:
1. May want to add that `These are aspects that are common to most software projects, open source or not.`
2. `a-Persistence` is TEAMMATES-specific
3. `a-Docs` may want to remove the "Website" reference because not all projects use that, or make it "User docs, e.g product website"